timemachine/fe/lambda_schedule.py

```diff
@@ -19,6 +22,18 @@ def interpolate_pre_optimized_protocol(pre_optimized_protocol, num_windows):
     return lambda_schedule
 
 
+def bisect_lambda_schedule(num_windows: int, lambda_interval: Tuple[float, float] = (0.0, 1.0)) -> NDArray[np.float64]:
+    """
+    Construct a lambda schedule with windows of size 2^N + 1 such that 2^N is the closest value to num_windows that is smaller.
+    This means that the number of windows returned may not match num_windows. Useful in the context of `run_sims_bisection`
+    where states are created on the fly with these same lambda values. Can save the cost of minimizing more often than necessary.
+    """
+    assert num_windows >= 2
+    min_lamb, max_lamb = lambda_interval
+    schedule_windows = int(2 ** np.floor(np.log2(num_windows))) + 1
```

Comment:
Small surprise: `len(schedule_windows) == 33` when `num_windows = 48`. Is this significant? Would we want this to be the same length as `num_windows`? (i.e. return the first `num_windows` points visited by this process https://github.com/proteneer/timemachine/pull/1347#discussion_r1685069111 ?)

Reply:
I don't think this is significant, unless we think that there is a lambda step size that we don't trust local minimization to perform well under. In this case any optimization step size is 1/33 (~0.03) which seems like a pretty small lambda step. If we think there need to be a certain number of initial states for this to be trustworthy, we probably should also have some warning of running with num_windows < threshold (Does num_windows == 3 become untrustworthy?)

Will run a hif2a graph to verify performance is unchanged.